### DISCUSSION TITLE
Plugin Development (Publisher vs Worker forms)

### DISCUSSION THREAD

**Discussion by rgdoms** [Keyfactor/ejbca-ce#682] [Q&A]
Hi.  
I have been developing some plugins for EJBCA, specifically adding custom publishers and custom workers, and I've run into a few challenges. I have several questions that I'll break down into different threads.  

- The publisher editing form includes a 'testConnection' method that gets called when you click 'Save and test connection.' It would be helpful to have a similar method that could be invoked when saving the worker, which could be overridden.  

- I attempted to intercept methods like 'save()' in the managedBean and also others in the EJB of PublisherSession to catch the deletion of a publisher, among other operations. I suspect that, for security reasons, this might not be feasible. I also tried using interceptors, ejb-jar.xml, and beans.xml. I think I might be missing something, but I thought it should be possible. Does anyone know if this can be done, and if so, how?  

Thanks

**Comment by primetomas** [comment]
1. I guess it's the same answer as for passwords. There is typically no "connection" from a worker, it just runs on the EJBCA instance.
I could see the benefit of something like a "run now" button, making it easy to test instead of having to wait for a timeout to trigger a run.

2. I'm not sure what you mean by intercept. the managed bean is a standard Java EE component, so it's bound by Java/Jakarta EE rules and processing. save() is called from the JSF framework from the JSF page where you click save/delete.
